ravenml/data/write_dataset.py

```diff
@@ -0,0 +1,166 @@
+import os, shutil, time, json
+import contextlib2
+from pathlib import Path
+from object_detection.dataset_tools import tf_record_creation_util
+from random import shuffle
+from datetime import datetime
+from ravenml.data.interfaces import CreateInput
+from ravenml.utils.question import cli_spinner, cli_spinner_wrapper
+
+class DatasetWriter(object):
+
+    def __init__(self, create: CreateInput, **kwargs):
+        self.associated_files = kwargs['associated_files']
+        self.related_data_prefixes = kwargs['related_data_prefixes']
+        
+        metadata = create.metadata
+        self.num_folds = metadata['kfolds'] if metadata.get('kfolds') else 5 # Never actually used
+        self.test_percent = metadata['test_percent'] if metadata.get('test_percent') else .2
+        self.label_to_int_dict = {}
+        self.dataset_path = create.dataset_path
+        self.dataset_name = metadata['dataset_name']
+        self.cache = create.plugin_cache
+    
+    def construct_all(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def export_as_TFExample(self, *args, **kwargs):
+        raise NotImplementedError
+    
+    def write_additional_files(self, *args, **kwargs):
```

Comment:
Decorators are not inherited from super classes, I messed around a bit with adding a metaclass in order to make it so the cli_spinner_wrapper wouldn't have to be set by the plugins but for now I am leaving it as is

Reply:
ideally we don't let plugins choose to use or not use spinner but instead centralize it and it does it automatically, but keeping it simple for now is best. definitely not urgent